tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -560,7 +560,7 @@ EOT
 	AGENT_DEST="${AGENT_DIR}/${AGENT_BIN}"
 
 	if [ -z "${AGENT_SOURCE_BIN}" ] ; then
-		[ "$ARCH" == "ppc64le" ] && ([ "$LIBC" == "gnu" ] || die "LIBC type for ppc64le should be gnu")
+		[ "$ARCH" == "ppc64le" ] && ([ "$LIBC" == "gnu" ] || (LIBC=gnu && echo "WARNING: LIBC type for ppc64le should be gnu"))
```

Comment:
This looks wrong to me - `( ... )` creates a sub-shell so changing `LIBC=` in that sub-shell won't affect the parent?!? In fact the only way the `die` is working currently is due to `set -e` faics.

Reply:
@jodh-intel, that's a nice catch!

Reply:
Thanks @jodh-intel @fidencio!
Shall the changes be dropped? or any suggestion?

Reply:
You need to use braces to modify the current shell. You could remove the test too and just force it:

```suggestion
		[ "$ARCH" == "ppc64le" ] &&  { LIBC=gnu; echo "WARNING: Forcing LIBC=gnu for ppc64le"; }
```

Yes, you'll then always get the warning, but that's probably good as it's a constant reminder that this (should be) a temporary fix until this can build on musl. Might be helpful to make that warning message specify a GitHub issue url to make it even clearer?

Reply:
Thanks @jodh-intel !

> Might be helpful to make that warning message specify a GitHub issue url to make it even clearer?

@bpradipt @Amulyam24 @clnperez
Is there an issue open for that?

Reply:
i dont believe there is @jing-wang4

Reply:
Open one?

Reply:
@jodh-intel Unfortunately I'm not aware of any plans for active musl toolchain support on Power. This might change in future but for now all focus is on the glibc toolchain. With this background, do you suggest we open a generic Kata issue to keep track of using musl with Kata on Power or make necessary changes in the Kata tool set to default to glibc for Power ?

Reply:
idm as long as it's clear from the warning message why we're setting glibc ;)

Reply:
@jodh-intel @bpradipt 
Please review the [updates](https://github.com/kata-containers/kata-containers/pull/1340/commits/71aeb920aac434ca4b21b3cf9275e33b2801c1a3) your feedback. Thanks!